src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs

```diff
@@ -79,10 +126,10 @@ internal static ThreeState GetObsoleteContextState(Symbol symbol, bool forceComp
         internal static DiagnosticInfo CreateObsoleteDiagnostic(Symbol symbol, BinderFlags location)
         {
             var data = symbol.ObsoleteAttributeData;
+            Debug.Assert(data != null);
```

Comment:
Why assert that data is not null, but then also test? Should this be `if (data == null) throw ExceptionUtilities.Unreachable;` instead?